store.go

```diff
@@ -676,6 +676,7 @@ type store struct {
 	autoNsMinSize   uint32
 	autoNsMaxSize   uint32
 	imageStore      rwImageStore
+	rwImageStores   []rwImageStore
```

Comment:
If we need to special-case this situation, fine.

This `rwImageStores`:
- has only zero or one element
- does not directly correspond to any user-visible UI feature

So to me, that’s an a surprising way to record the data. It seems to me to be an unnecessary generalization just complicating the code right now.

_At the very least_ this field needs documentation as to how it relates to `imageStore`/`roImageStores` (it _looks_ as if it is some kind of direct sibling to `imageStore`/`roImageStore`, which is not the case). And I’d prefer the relevant stores to be somehow directly modeled here — “primary image store” / “graph-root image store” / “additional image stores as configured by user”, or something like that.

---

To my way of thinking, it’s surprising, inconsistent and hard to follow that the responsibility for `options.ImageStore` is so diffuse. This `store` object is responsible for creating the directories based on `options.ImageStore`, but somehow the individual graph driver is responsible for turning `options.ImageStore` into an entry in `AdditionalImageStores`, but it _must_ use precisely the `s.graphRoot` path, or the upper-level `store` breaks.

This is not at all a clean division of responsibility between the top-level store and the graph driver. I feel rather confident in predicting that it will lead to bugs.

(To be fair, all of the “additional * store” has similar design entanglement between `store` and the overlay driver; but this is expanding it to a much larger degree.)

---

Either this whole image store feature is a UI convenience over the existing “additional read-only stores” feature, and in that case it should exist ~entirely in the top-level objects close to the UI. (I think this should be the case, but that’s not true after this PR.) And then the graph driver should be _completely unaware_ of the `options.ImageStore` option.

Or it is really a cross-cutting feature across the stores and the drivers, and in that case I think it should be _explicitly modeled_ as Go struct fields, and methods that can fail. The top-level store should, at the very least, _know_ whether the graph driver implements the “graphRoot additional image store” (maybe by the driver implementing a method that returns a value about that). And then, again, the store can just instruct the driver to use an additional image store, without the driver explicitly checking `options.ImageStore`.

It’s a strong “smell” of unclean design to me that `store` and the drivers (only one of them!), within the same codebase, communicate via `fmt.Sprintf("AdditionalImageStore=%s", options.ImageStore)`. Why use text when we we have Go struct fields?